### PR TITLE
Add TON NFT support

### LINKED ISSUE #137: Add TON NFT's support
- [ ] https://github.com/gemwalletcom/core/issues/136 (needs to be implemented first)

Use https://docs.tonconsole.com/tonapi/api-v2 to return NFT's

### PR DESCRIPTION
**Pull request by kehiy**
- Fixes #137

### CODE REVIEW

File: crates/gem_chain_rpc/src/ton/client.rs [hunk 1]
```diff
@@ -60,6 +60,17 @@ impl TonClient {
         None
     }
 
+    pub fn map_nft(&self, nft: Nft) -> Option<primitives::NFT> {
+        let nft = primitives::NFT::new(
+            primitives::chain::Chain::Ton,
```
Comment (gemcoder21) [inline crates/gem_chain_rpc/src/ton/client.rs]: `primitives::chain::Chain::Ton` => `self.get_chain()`

File: crates/gem_chain_rpc/src/lib.rs [hunk 2]
```diff
@@ -55,3 +55,24 @@ where
         (**self).get_transactions(block_number).await
     }
 }
+
+#[async_trait]
+pub trait ChainNFTProvider: Send + Sync {
+    async fn get_collectibles(
```
Comment (gemcoder21) [inline crates/gem_chain_rpc/src/lib.rs:61]: let's add get collections method here too

Comment (gemcoder21) [reply crates/gem_chain_rpc/src/lib.rs:61]: collections would be the first endpoint to use and then request collectibles

Comment (kehiy) [reply crates/gem_chain_rpc/src/lib.rs:61]: Okay, I'll add it to the trait. but for TON, in ton API can't find the endpoint to get collections that a user owns. we have 2 solutions:
1. keep it unimplemented for TON.
2. Get all user NFTs search for their collection and return them.

**Pull request closed without merging**